### PR TITLE
TSDB: Make time boundaries settings required in tsdb indices

### PR DESCRIPTION
**Pull request by weizijun**
Make `index.time_series.start_time` and `index.time_series.end_time` settings as required settings in tsdb indices.

It will change many tests, among which a time_series index is created, and it will add the `index.time_series.start_time` and `index.time_series.end_time` settings

### CODE REVIEW

**Comment by weizijun** [comment]
@elasticsearchmachine  update branch

**Review by nik9000 — COMMENTED** [review]
I think it'd be nice to  use times in the tests that are a little closer to the test data. If we're really make truly random times from 1 to Long.MAX_VALUE it's cool to have those as bounds, but for the tests I'd feel better with bounds that are a day or a month long or something like that.

**Comment by nik9000** [comment]
@elasticmachine, test this please

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/tsdb/60_add_dimensions.yml [hunk 1]
```diff
@@ -114,8 +114,8 @@ add dimensions to no dims with dynamic_template over bulk:
                 mode: time_series
                 routing_path: [metricset]
                 time_series:
-                  start_time: 1
-                  end_time: 253402300799998
+                  start_time: 2021-04-28T00:00:00Z
+                  end_time: 2999-04-28T00:00:00Z
```
Comment (nik9000) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/tsdb/60_add_dimensions.yml:118]: Maybe use something closer to the range of data we have?

Comment (weizijun) [reply rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/tsdb/60_add_dimensions.yml]: done, and there left some failed tests, I will run gradle check to find all failed tests, and fix it tomorrow

**Comment by weizijun** [comment]
@nik9000 can I change all tsdb yaml test's skip config from:
```
      version: " - 7.99.99"
      reason: introduced in 8.0.0
```
to
```
      version: " - 8.0.99"
      reason: introduced in 8.1.0
```
to pass the mixed-cluster tests

**Comment by nik9000** [comment]
> to pass the mixed-cluster tests

:+1:

**Comment by nik9000** [comment]
@elasticmachine, test this please

**Comment by weizijun** [comment]
@elasticmachine update branch

**Comment by nik9000** [comment]
run elasticsearch-ci/release-tests

**Comment by nik9000** [comment]
> @nik9000 the failed elasticsearch-ci/release-tests seems that the problem of the environment

:+1:. Let's see if it's better a second time.

**Comment by nik9000** [comment]
run elasticsearch-ci/release-tests

**Comment by nik9000** [comment]
@elasticmachine, update branch

**Pull request merged**